### ISSUE TITLE
IllegalArgumentException Invalid target position when swipe-scrolling and using month navi buttons

### ISSUE THREAD

**Issue by laalto** [wdullaer/MaterialDateTimePicker#512]
Repro steps:

1. Launch the date picker dialog with minimum date set to today
2. Go to next month in the picker
3. Swipe to scroll back to current month
4. While the recyclerview is still scrolling, tap on the "previous month" button.

Crash:

```
Fatal Exception: java.lang.IllegalArgumentException
Invalid target position
android.support.v7.widget.RecyclerView$LayoutManager.android.support.v7.widget.RecyclerView$SmoothScroller.start (SourceFile:14377)
android.support.v7.widget.LinearLayoutManager.smoothScrollToPosition$7d69765f (SourceFile:453)
android.support.v7.widget.RecyclerView.smoothScrollToPosition (SourceFile:1700)
com.wdullaer.materialdatetimepicker.date.DayPickerGroup.onClick (SourceFile:168)
```

Looks like `DayPickerGroup.onClick()` should validate the computed position before attempting to scroll to it - `updateButtonVisibility()` does not disable the arrow buttons immediately on swipe.

Likely there's a similar problem with a maximum date / "next month" but I did not try to reproduce with that scenario.

**Comment by laalto** [comment]
Looks like this problem is still there. Can repro this issue on an app with MDTP 3.7.4 with the following stacktrace:

```
Fatal Exception: java.lang.IllegalArgumentException: Invalid target position
       at android.support.v7.widget.RecyclerView$LayoutManager.android.support.v7.widget.RecyclerView$SmoothScroller.start(SourceFile:14534)
       at android.support.v7.widget.LinearLayoutManager.smoothScrollToPosition$7d69765f(SourceFile:451)
       at android.support.v7.widget.RecyclerView.smoothScrollToPosition(SourceFile:1781)
       at com.wdullaer.materialdatetimepicker.date.DayPickerGroup.onClick(SourceFile:183)
```

**Comment by wdullaer** [comment]
I can reproduce this, but don't understand it yet.

**Comment by wdullaer** [comment]
This is now properly fixed in 4.0.1
Sorry for the inconvenience.
